File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.py
n, k = map(int, input().split())  # 1부터 n까지, k번쨰사람을 제거
arr = [i for i in range(1, n + 1)]
stack = []  # 출력할 대상을 넣을 배열
idx = 0

for i in range(n):
    idx += k - 1  # 큐에 있는 rear를 k번으로 이동
    if idx >= len(arr):  # rear이 arr범위를 넘어간다면
        idx = idx % len(arr)  # 그 길이만큼 나눈 나머지가 rear이 됨!!(중요)
    stack.append(str(arr.pop(idx)))  # 답을 넣기

print('<', ', '.join(stack), '>', sep='')  # 구분자로 띄어쓰기 없애기

File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.java


import java.io.*;
import java.util.*;

public class Main {
	
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String arr = br.readLine();
		StringBuilder sb = new StringBuilder();
		int n = Integer.parseInt(arr.split(" ")[0]);
		int k = Integer.parseInt(arr.split(" ")[1]);
		List<Integer> list = new LinkedList<>();
		for (int i=1; i<n+1; i++) {
			list.add(i);
		}
		sb.append("<");
		int idx=0;
		while (!list.isEmpty()) {
			if (list.size()==1) {
				idx = (idx+(k-1))%(list.size());
				sb.append(list.get(idx));
				list.remove(idx);
			}
			else {
			idx = (idx+(k-1))%(list.size());
			sb.append(list.get(idx)).append(",").append(" ");
			list.remove(idx);
			}
		}
		sb.append(">");
		System.out.println(sb);
		
	}
}


	